diff --git a/tools/mtmd/CMakeLists.txt b/tools/mtmd/CMakeLists.txt
--- a/tools/mtmd/CMakeLists.txt
+++ b/tools/mtmd/CMakeLists.txt
@@ -89,7 +89,7 @@ set_target_properties(mtmd
     PROPERTIES
     PRIVATE_HEADER debug/mtmd-debug.h)
 
-install(TARGETS mtmd LIBRARY PUBLIC_HEADER)
+install(TARGETS mtmd LIBRARY ARCHIVE PUBLIC_HEADER)
 
 if (NOT MSVC)
     # for stb_image.h and miniaudio.h
@@ -103,7 +103,6 @@ endif()
 
 if (TARGET BUILD_INFO)
     add_dependencies(mtmd        BUILD_INFO)
-    add_dependencies(mtmd-helper BUILD_INFO)
 endif()
 
 # if mtmd is linked against llama-common, we throw an error
