diff --git a/CMakeLists.txt b/CMakeLists.txt
index b6e4bb4..acae85f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -248,7 +248,6 @@ function (setup_target TARGET_NAME)
                     -Wextra
                     -Wno-conversion
                     -Wno-unknown-pragmas
-                    -march=native
                     -fmax-errors=1
                     -pedantic
                     -fdiagnostics-color=always
diff --git a/cpp/bench.cpp b/cpp/bench.cpp
index c04edab..3d3adeb 100644
--- a/cpp/bench.cpp
+++ b/cpp/bench.cpp
@@ -35,6 +35,7 @@
 #include <stdexcept>     // `std::invalid_argument`
 #include <string>        // `std::to_string`
 #include <thread>        // `std::thread::hardware_concurrency()`
+#include <type_traits>   // `std::result_of` in clipp
 #include <unordered_map> // `std::unordered_map`
 #include <variant>       // `std::monostate`
 #include <vector>        // `std::vector`
diff --git a/cpp/test.cpp b/cpp/test.cpp
index 1ab4f06..b7afe12 100644
--- a/cpp/test.cpp
+++ b/cpp/test.cpp
@@ -22,6 +22,7 @@
 #include <algorithm>     // `std::shuffle`
 #include <random>        // `std::default_random_engine`
 #include <stdexcept>     // `std::terminate`
+#include <type_traits>   // `std::add_const` in StringZilla
 #include <unordered_map> // `std::unordered_map`
 #include <vector>        // `std::vector`

diff --git a/include/usearch/index.hpp b/include/usearch/index.hpp
index eeb18f5..2105ab6 100644
--- a/include/usearch/index.hpp
+++ b/include/usearch/index.hpp
@@ -94,11 +94,13 @@
 #include <climits>   // `CHAR_BIT`
 #include <cmath>     // `std::sqrt`
 #include <cstring>   // `std::memset`
+#include <exception> // `std::uncaught_exceptions`
 #include <iterator>  // `std::reverse_iterator`
 #include <mutex>     // `std::unique_lock` - replacement candidate
 #include <random>    // `std::default_random_engine` - replacement candidate
 #include <stdexcept> // `std::runtime_exception`
 #include <thread>    // `std::thread`
+#include <type_traits> // `std::is_pod`, `std::enable_if`
 #include <utility>   // `std::pair`

 // Helper macros for concatenation and stringification
